x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/FastOnePoisson.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs;
+
+import java.util.Arrays;
+
+public class FastOnePoisson {
+
+    private static final double[] CDF;
+    private static final double[] COND_CDF;
+    private static final int[] BUCKET_SAMPLES;
+    private static final double[] CORRECTIONS;
+    private static final int NEED_TO_GENERATE = 4;
+    private static final double RNG_RANGE = 4294967296.0; // = 2^32
+    static {
+        CDF = new double[12];
+        double f = Math.exp(-1.0);
+        CDF[0] = f;
+        for (int i = 1; i < 12; i++) {
+            f /= (i);
```

Comment:
nit: omit paretheses (here and in similar places)?

Reply:
++